ui/admin/app/controllers/scopes/scope/targets/index.js

```diff
@@ -91,4 +97,225 @@ export default class ScopesScopeTargetsIndexController extends Controller {
+        { hostSourceCount },
+      );
+
+      // Ask for confirmation
+      await this.confirm.confirm(confirmMessage, {
+        title: 'resources.target.questions.delete-host-sources.title',
+        confirm: 'actions.remove-resources',
+      });
+      // Remove host sources.  This step is reached only if the user accepts.
+      await target.removeHostSources(hostSourceIDs);
+    }
+  }
+
+  /**
+   * Deletes a target and redirects to targets index.
+   * @param {TargetModel} target
+   */
+  @action
+  @loading
+  @confirm('questions.delete-confirm')
+  @notifyError(({ message }) => message, { catch: true })
+  @notifySuccess('notifications.delete-success')
+  async delete(target) {
+    await target.destroyRecord();
+    this.router.replaceWith('scopes.scope.targets');
+    await this.router.refresh();
+  }
+
+  /**
+   * Update type of target
+   * @param {string} type
+   */
+  @action
+  changeType(type) {
+    this.router.replaceWith({ queryParams: { type } });
+  }
+
+  /**
+   * Remove a credential source from the current target.
+   * @param {TargetModel} target
+   * @param {CredentialLibraryModel, credentialModel} credentialSource
+   */
+  @action
+  @loading
+  @confirm('questions.remove-confirm')
+  @notifyError(({ message }) => message, { catch: true })
+  @notifySuccess('notifications.remove-success')
+  async removeInjectedApplicationCredentialSource(target, credentialSource) {
+    await target.removeInjectedApplicationCredentialSource(credentialSource.id);
+    await this.router.refresh();
+  }
+
+  /**
+   * Remove a credential source from the current target.
+   * @param {TargetModel} target
+   * @param {CredentialLibraryModel, credentialModel} credentialSource
+   */
+  @action
+  @loading
+  @confirm('questions.remove-confirm')
+  @notifyError(({ message }) => message, { catch: true })
+  @notifySuccess('notifications.remove-success')
+  async removeBrokeredCredentialSource(target, credentialSource) {
+    await target.removeBrokeredCredentialSource(credentialSource.id);
+    await this.router.refresh();
+  }
+
+  /**
+   * Removes a host set from the current target and redirects to index.
+   * @param {TargetModel} target
+   * @param {HostSetModel} hostSet
+   */
+  @action
+  @loading
+  @confirm('questions.remove-confirm')
+  @notifyError(({ message }) => message, { catch: true })
+  @notifySuccess('notifications.remove-success')
+  async removeHostSource(target, hostSet) {
+    await target.removeHostSource(hostSet.id);
+    await this.router.refresh();
+  }
+
+  /**
+   * Delete an alias
+   * @param {AliasModel} alias
+   */
+  @action
+  @loading
+  @confirm('resources.alias.messages.delete')
+  @notifyError(({ message }) => message, { catch: true })
+  @notifySuccess('notifications.delete-success')
+  async deleteAlias(alias) {
+    await alias.destroyRecord();
+    await this.router.transitionTo('scopes.scope.targets.target');
+    await this.router.refresh('scopes.scope.targets.target');
+  }
```

Comment:
crudl actions for aliases exist in this controller since they are used in manage-alias and create-alias routes.